File: code/C++/58-Length of Last Word .cpp
/*
Length of Last Word
Given a string s consists of upper/lower-case alphabets and empty space characters ' ', return the length of last word in the string.

If the last word does not exist, return 0.

Note: A word is defined as a character sequence consists of non-space characters only.

For example, 
Given s = "Hello World",
return 5.
逻辑上从前往后扫描简单一点，但是我不想做很多无用计算，就从后往前扫了，所以会有很多细节要考虑
*/

class Solution {
public:
    int lengthOfLastWord(const char *s) {
        int len=strlen(s);
        
        
        while(len>=0&&s[len-1]==' ')
          len--;
        if(len<=0)
          return 0; 
          
          
        int i=1;
        
        char c=s[len-i];
        for(;i<=len&&c!=' ';i++){
            c=s[len-i];
        }
        
        if(c==' ')
         return i-2;
        else
         return i-1;
          
        
    }
};

File: code/Python/58-Length of Last Word.py
# Given a string s consists of upper/lower-case alphabets and empty space characters ' ', return the length of last word (last word means the last appearing word if we loop from left to right) in the string.
#
# If the last word does not exist, return 0.
#
# Note: A word is defined as a maximal substring consisting of non-space characters only.
#
# Example:
#
# Input: "Hello World"
# Output: 5
#

class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        while len(s) != 0 and s[-1] == ' ':
            s = s[:-1]
        return len(s.split(' ')[-1])

    
if __name__ == '__main__':
    s = Solution()
    s.lengthOfLastWord("a")